lib/promoted/ruby/client.rb

```diff
@@ -102,18 +103,18 @@ def deliver args, headers={}
               rescue  StandardError => err
                 # Currently we don't propagate errors to the SDK caller, but rather default to returning
                 # the request insertions.
+                deliver_err = true
                 @logger.error("Error calling delivery: " + err.message) if @logger
```

Comment:
side-comment - in promoted-ts-client, this is a handleError that can do whatever it wants (including throw).  JFYI.  It's behavior is a little different.

Reply:
Ah, looking at the code, the scope of the try/catch is different so these conditionals are a little diffent.

Reply:
If it makes sense to keep them in sync, feel free to change one or the other.

Reply:
I don't think there's anything in the broader scope that can raise. The "handleError" approach isn't very Ruby-like, which is why I was trying to avoid it.